internal/auth0/organization/resource_member_role.go

```diff
@@ -0,0 +1,134 @@
+package organization
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/auth0/go-auth0/management"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/auth0/terraform-provider-auth0/internal/config"
+	internalSchema "github.com/auth0/terraform-provider-auth0/internal/schema"
+)
+
+// NewMemberRoleResource will return a new auth0_organization_member_role (1:1) resource.
+func NewMemberRoleResource() *schema.Resource {
+	return &schema.Resource{
+		Description:   "This resource is used to manage the roles assigned to an organization member.",
+		CreateContext: createOrganizationMemberRole,
+		ReadContext:   readOrganizationMemberRole,
+		DeleteContext: deleteOrganizationMemberRole,
+		Importer: &schema.ResourceImporter{
+			StateContext: internalSchema.ImportResourcePairID("organization_id", "user_id"),
+		},
+		Schema: map[string]*schema.Schema{
+			"organization_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The ID of the organization.",
+			},
+			"user_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The user ID of the organization member.",
+			},
+			"role_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The role ID to assign to the organization member.",
+			},
+			"role_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Name of the role.",
+			},
+			"role_description": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Description of the role.",
+			},
+		},
+	}
+}
+
+func createOrganizationMemberRole(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+	mutex := meta.(*config.Config).GetMutex()
+
+	organizationID := data.Get("organization_id").(string)
+	userID := data.Get("user_id").(string)
+	roleID := data.Get("role_id").(string)
+
+	mutex.Lock(organizationID)
+	defer mutex.Unlock(organizationID)
+
+	if err := api.Organization.AssignMemberRoles(organizationID, userID, []string{roleID}); err != nil {
+		if err, ok := err.(management.Error); ok && err.Status() == http.StatusNotFound {
+			return nil
+		}
+
+		return diag.FromErr(err)
+	}
+
+	data.SetId(organizationID + ":" + userID)
```

Comment:
Need to attach the role ID to this too to enable management of multiple roles for a org member.